### PR TITLE
(chore) Bump i18next, react-i18next and i18next-parser and support pluralization

### PR DESCRIPTION
**Pull request by vasharma05**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary
This PR upgrades the i18next related packages. The changes in the new upgrades are as follows:

### Changes in i18next from v19 to v21
JSON V4 Support: https://www.i18next.com/misc/json-format#i-18-next-json-v4
Pluralization support: https://www.i18next.com/translation-function/plurals

### Changes in react-i18next parser
No major changes as per documentation.

### i18next-parser from v5 → v6
Node v ≤ 14 not supported anymore
Deprecated skipDefaultValues and useKeysAsDefaultValues options

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -4,4 +4,4 @@
 set -e  # die on error
 
 yarn prettier && npx lint-staged
-yarn turbo run extract-translations
+yarn run extract-translations
```
Comment (vasharma05) [inline .husky/pre-commit:7]: This is the only repo using turbo in extract-translations, added the script in package.json using lerna, to conform with other repos.

CC: @denniskigen @ibacher

File: package.json [hunk 2]
```diff
@@ -13,7 +13,8 @@
     "prettier": "prettier --config prettier.config.js --write \"packages/**/*.{ts,tsx,css,scss}\" \"e2e/**/*.ts\"",
     "postinstall": "husky install",
     "ci:bump-form-engine-lib": "yarn up @openmrs/openmrs-form-engine-lib@next",
-    "test-e2e": "playwright test"
+    "test-e2e": "playwright test",
+    "extract-translations": "lerna run extract-translations -- --config ../../tools/i18next-parser.config.js"
```
Comment (vasharma05) [inline package.json:17]: @denniskigen @ibacher

File: tools/i18next-parser.config.js [hunk 3]
```diff
@@ -65,15 +65,6 @@ module.exports = {
...
 
-  skipDefaultValues: false,
-  // Whether to ignore default values
-  // You may also specify a function accepting the locale and namespace as arguments
-
-  useKeysAsDefaultValue: false,
-  // Whether to use the keys as the default value; ex. "Hello": "Hello", "World": "World"
-  // This option takes precedence over the `defaultValue` and `skipDefaultValues` options
-  // You may also specify a function accepting the locale and namespace as arguments
-
```
Comment (vasharma05) [inline tools/i18next-parser.config.js:76]: Deprecated

File: .husky/pre-commit (hunk 1 above)
Comment (ibacher) [reply .husky/pre-commit:7]: We're intentionally using Turbo instead of Lerna. Please undo this (fine to do it via script, just use Turbo).

Comment (vasharma05) [reply .husky/pre-commit:7]: Sure @ibacher !

**Review by jayasanka-sack — APPROVED** [review]
Thanks for getting this done, Vineet!

**Pull request merged**